### DISCUSSION TITLE
绘制的图形删除事件在哪里

### DISCUSSION THREAD

**Discussion by lisuai** [zouyaoji/vue-cesium#445] [Q&A] [answered]
绘制的图形删除事件在哪里
![img_v2_f409e87d-a62d-4ff4-a45b-e5a9bfe6415g](https://github.com/zouyaoji/vue-cesium/assets/68360102/58fd2b97-cab2-40b4-a181-2093a1fe132e)

**Comment by zouyaoji** [comment, marked as answer]
你好，是`editorEvt`

**Reply by PetukhovArt** [reply]
how i can call stop event of action Instance from client? 
to stop drawing some primitive on escape button

**Reply by PetukhovArt** [reply]
omg , i found how to call stop(), but it not working correctly ( the tip not cancelling)
btw action instance has some public methods, that are not described in Typescript , so this are the additional type of action instance cmpRef : 
![image](https://github.com/user-attachments/assets/8016950f-390a-4107-aafd-49b18a0daf0e)
 
P.S. 
so it was easy to manually delete current drawing action result :
`drawingsRef.value.toggleAction(some actionName)`

**Reply by zouyaoji** [reply]
https://github.com/zouyaoji/vue-cesium/blob/dev/packages/composables/use-drawing/use-drawing-polyline.ts#L355


You can use ref template references in Vue to get access to each drawing tool's stop method. Here's an example:
![image](https://github.com/user-attachments/assets/72eb9cb4-ea6c-43a3-a787-c1c6160d3cb4)

**Reply by NanF007** [reply]
@zouyaoji 大佬！求指点一下🙏
<img width="425" height="256" alt="ceshi1" src="https://github.com/user-attachments/assets/8e6456a5-8524-4bd3-b081-34a5c9a2963e" />
请问怎么才能开启点位时能实现下图效果啊
<img width="602" height="249" alt="ceshi2" src="https://github.com/user-attachments/assets/e544d2bf-c409-4610-afe2-3cdbef05948a" />
下面这张图逻辑应该是鼠标悬浮顶点可直接拖拽改变位置，悬浮至中间点可拖拽后新增顶点，我现在做的这个用的是vc-drawings进行多边形编辑，但是我发现只有顶点，没有中间点，且移动、新增、移除单个顶点都只能先悬浮触发按钮菜单再选择一个后才可做到移动、增加、移除
